src/template.ts

```diff
@@ -0,0 +1,152 @@
+  exports: string;
+}
+
+export default function gbTemplate(
+  { template }: { template: templateType },
+  opts: { typescript: object },
+  { imports, interfaces, componentName, props, jsx, exports }: astPartsType,
+) {
+  const plugins: ParserPlugin[] = ['jsx'];
+
+  if (opts.typescript) {
+    plugins.push('typescript');
+  }
+
+  const hexRegex = RegExp('^#(?:[0-9a-fA-F]{3}){1,2}$');
+
+  const isHex = (value: string) => hexRegex.test(value);
+
+  const isFillStrokeOrStopColor = (name: string | JSXIdentifier) =>
+    name === 'fill' || name === 'stroke' || name === 'stopColor';
+
+  const getColorsAndModifyAttr = (coCount: number, attributes: Array<any>) => {
+    let colorArray: Array<string> = [];
+
+    attributes.map((attr) => {
+      if (attr.value! && isFillStrokeOrStopColor(attr.name.name) && isHex(attr.value.value)) {
+        colorArray.push(attr.value.value);
+        attr.value.type = 'JSXExpressionContainer';
+        attr.value.expression = { type: 'Identifier', name: `colors[${coCount}]` };
+        coCount++;
+      }
+      return attr;
+    });
+    return { coCount, colorArray };
+  };
+
+  let count = 0;
+  let finalColors: Array<string> = [];
+  const loopJsxChildrenAndBuildAttr = (
+    children: (JSXElement | JSXText | JSXExpressionContainer | JSXSpreadChild | JSXFragment)[],
+  ) => {
+    children.forEach((childElement: any) => {
+      if (childElement.openingElement.attributes.length > 0) {
+        const { coCount, colorArray } = getColorsAndModifyAttr(
+          count,
+          childElement.openingElement.attributes,
+        );
+        count = coCount;
+        finalColors.push(...colorArray);
+      }
+      if (childElement.children.length > 0) {
+        loopJsxChildrenAndBuildAttr(childElement.children);
+      }
+    });
+  };
+
+  const finalJsx = JSON.parse(JSON.stringify(jsx));
+
+  loopJsxChildrenAndBuildAttr(finalJsx.children);
+
+  const babelTypeColors = finalColors.map((color) => {
+    return {
+      type: 'StringLiteral',
+      value: color,
+    };
+  });
+
+  const defaultColorsJsx = [
+    {
+      type: 'VariableDeclaration',
+      declarations: [
+        {
+          type: 'VariableDeclarator',
+          id: {
+            type: 'Identifier',
+            name: 'defaultColors',
+          },
+          init: {
+            type: 'ArrayExpression',
+            elements: babelTypeColors,
+          },
+        },
+      ],
+      kind: 'const',
+    },
+  ];
+
+  const propInterface = [
+    interfaceDeclaration(
+      identifier('componentPropsInterface'),
+      null,
+      [],
+      objectTypeAnnotation([
+        objectTypeProperty(identifier('colors'), arrayTypeAnnotation(stringTypeAnnotation())),
+        objectTypeProperty(identifier('props'), genericTypeAnnotation(identifier('SvgProps'))),
+      ]),
+    ),
+  ];
+
+  const destructuredProps = props.map((item) => {
+    item.name = '{ colors = defaultColors, ...props }: componentPropsInterface';
+    return item;
+  });
```

Comment:
I noticed a small issue with the previous type definition converting semi-colons to commas after generating the `jsCode` so now we are:

- Declaring an interface using babel types api